client-cli/src/command.rs

```diff
@@ -193,6 +198,12 @@ pub enum Command {
             help = "Disable fast forward, which is not secure when connecting to outside nodes"
         )]
         disable_fast_forward: bool,
+        #[structopt(
+            name = "disable-address-recovery",
+            long,
+            help = "Disable address recovery, which is not necessary, if addresses already exist"
```

Comment:
should there be a default value?

Reply:
i'll check with mock hardware wallet

Reply:
default is using "address recovery".

Reply:
But "HW wallet" will be a "HD wallet" as well -- the difference is that it doesn't expose you private information -- I assume one would want the history recovery as well... open an issue for it?

Reply:
"HW" wallet is generally also a "HD wallet" -- it's that it doesn't expose private keys and the current code relies on having private keys.
One can derive the public keys / addresses from the parent public key: https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#public-parent-key--public-child-key